scripts/Inelastic/Direct/DirectEnergyConversion.py

```diff
@@ -1293,6 +1294,8 @@ def save_results(self, workspace, save_file=None, formats=None):
                     SaveSPE(InputWorkspace=workspace,Filename= filename)
                     break
                 if case('nxs'):
+                    if save_file[-4:] != '_red':
+                        save_file = save_file+'_red'
```

Comment:
Could we add a new case instead over overloading the `nxs` case - this might be used for normal (non-indirect) reduction too.
```suggestion
                if case('indirect'):
                    if save_file[-4:] != '_red':
                        save_file = save_file+'_red'
                if case('nxs'):
```
(does your `switch` implementation allow a case fall through?)

Reply:
@mducle This would not work -- we reduce direct, and to be accepted by indirect analysis without manual renaming, it would request `_red` suffix. We are not currently using nxs in direct analysis, so no agreement on the file format/name is reached/requested As source files are nxs and the reduced files are nxs, having `_red` suffix to highligt processed files looks reasonable.

Reply:
Yeah, I understand the need for the indirect interface to have `_red` but I'd like this to be a special case of `save_format='indirect'` rather than `save_format='nxs'` - in any case there should not be a name collision because the reduced data files we generate have the an Ei suffix whereas the raw data file is just instrument name + run number.

> No indirect case exist in our reduction as we reduce direct data

It's just a label - it doesn't have to correspond to a file extension or anything. I'm just saying that we create a new label. That way, if you specify `iliad(..., save_format='indirect')` (or change the reduction file property to `prop['save_format'] = 'indirect'`) it will generate an `_red.nxs` and otherwise it would just generate a un-suffixed `nxs`.

Reply:
PS. If you create a new label you also don't need to change the unit tests ;)

Reply:
PPS. 

> We are not currently using nxs in direct analysis

Yes, we *at ISIS* are not using `nxs` but I don't know what SNS are doing... 

Also, we may end up using it in future - I personally would prefer to ditch `nxspe` and move to `nxs` in Horace since there is much more metadata in the `nxs` files which would be useful (e.g. sample environment log values). As such I think we should preserve the `nxs` label for future direct analysis use.

Reply:
I do not know and do not care very much but let's invite Tatiana to this discussion. I feel `indirect` key would be confusing for users as the files are direct. And the agreement that `indirect` means just `_red` extension is a bit convoluted and "indirect" knowledge. Why not to adopt convention, already existing in other groups.... I would rather requested 'red' which would produce `_red.nxs`

Tests have to be written and modified in any case -- for new keyword too.